cmd/notation/internal/display/metadata/json/inspect.go

```diff
@@ -0,0 +1,202 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package json
+
+import (
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/hex"
+	"fmt"
+	"strings"
+	"time"
+
+	coresignature "github.com/notaryproject/notation-core-go/signature"
+	"github.com/notaryproject/notation-go/plugin/proto"
+	"github.com/notaryproject/notation/cmd/notation/internal/display/output"
+	"github.com/notaryproject/notation/internal/envelope"
+	"github.com/notaryproject/tspclient-go"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type inspectOutput struct {
+	MediaType  string       `json:"mediaType"`
+	Signatures []*signature `json:"signatures"`
+}
+
+// signature is the signature envelope for printing in JSON format.
+type signature struct {
+	Digest                string             `json:"digest,omitempty"`
+	SignatureAlgorithm    string             `json:"signatureAlgorithm"`
+	SignedAttributes      map[string]any     `json:"signedAttributes"`
+	UserDefinedAttributes map[string]string  `json:"userDefinedAttributes"`
+	UnsignedAttributes    map[string]any     `json:"unsignedAttributes"`
+	Certificates          []*certificate     `json:"certificates"`
+	SignedArtifact        ocispec.Descriptor `json:"signedArtifact"`
+}
+
+// certificate is the certificate information for printing in JSON format.
+type certificate struct {
+	SHA256Fingerprint string    `json:"SHA256Fingerprint"`
+	IssuedTo          string    `json:"issuedTo"`
+	IssuedBy          string    `json:"issuedBy"`
+	Expiry            time.Time `json:"expiry"`
+}
+
+// timestamp is the timestamp information for printing in JSON format.
+type timestamp struct {
+	Timestamp    string         `json:"timestamp,omitempty"`
+	Certificates []*certificate `json:"certificates,omitempty"`
+	Error        string         `json:"error,omitempty"`
+}
+
+// InspectHandler is the handler for inspecting metadata information and
+// rendering it in JSON format. It implements the metadata.InspectHandler
+// interface.
+type InspectHandler struct {
+	printer *output.Printer
+
+	output inspectOutput
+}
+
+// NewInspectHandler creates a new JsonHandler to inspect signatures and print in
+// JSON format.
+func NewInspectHandler(printer *output.Printer) *InspectHandler {
```

Comment:
The package name is `json` instead of `metadata`, so how would a user know this method is metadata specific?
```suggestion
// NewMetadataInspectHandler creates a new metadata JsonHandler to inspect signatures and print in
// JSON format.
func NewMetadataInspectHandler(printer *output.Printer) *InspectHandler {
```

Reply:
The package is under `metadata` package, which means all handler structs are for metadata. Can we keep it simple with the original name?

Reply:
> The package is under `metadata` package, which means all handler structs are for metadata.

This doesn't really make sense in Golang? The package's name is `json`, which means, callers will use `json.NewInspectHandler` in their code. In case of a command supports multiple JSON display types, the current naming is not clear enough.

Reply:
JSON and tree are only for metadata type. I think it is too redundant, and every other struct and constructor under metadata/* will have a Metadata prefix.